plugin/PDFView.qml

```diff
@@ -214,17 +220,29 @@ SilicaFlickable {
                 }
 
                 canvas: pdfCanvas
+                selection: pdfSelection
+
+                onPositionChanged: contextHook.positionChanged(Qt.point(at.x - contentX,
+                                                                        at.y - contentY))
+
                 onLinkClicked: base.linkClicked(linkTarget, contextHook)
                 onGotoClicked: base.goToPage(page - 1, top, left,
                                              Theme.paddingLarge, Theme.paddingLarge)
-                onClicked: {
-                    if (selection.text.length > 0) {
-                        selection.unselect()
-                    } else {
-                        base.clicked()
-                    }
+                onSelectionClicked: base.selectionClicked(selection, contextHook)
+                onAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onTextAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onCaretAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onHighlightAnnotationClicked: base.annotationClicked(annotation, contextHook)
+                onClicked: base.clicked()
+                onAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
+                onTextAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
+                onCaretAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
+                onHighlightAnnotationLongPress: base.annotationLongPress(annotation, contextHook)
```

Comment:
I've tested the following, but it didn't work. Maybe you can spot a mistake somewhere.
pdflinkarea.h:
signal annotationPress(PDFAnnotation *anno);

pdflinkarea.cpp:
        switch (d->annotation->subType()) {
        case Poppler::Annotation::SubType::AText: {
            PDFTextAnnotation *proxy = new PDFTextAnnotation(…);
            emit annotationPress(static_cast<PDFAnnotation*>(proxy));
            return;
        }
…

PDFDocumentPage.qml:
onAnnotationPress: if (anno.type == PDF.Annotation.Text) console.log(anno.icon)

This raises a runtime error, saying that icon is not a property of type PDFAnnotation. So what I've deduced is that, having a signal with signature (Annotation *anno), make the QML part loose the information that this anno object is indeed a specialized text version of annotation. Thus, I decided to create as many signals with detailled signature as I have of specialized types, then I store all these different QML object into a single variant QML property, so the type is preserved even if accessed from the same name.